particula/next/gas/vapor_pressure.py

```diff
@@ -0,0 +1,370 @@
+        """
+        Calculate the vapor pressure at a new temperature using the
+        Clausius-Clapeyron equation. For ideal gases at low temperatures.
+
+        Args:
+        ----
+        - temperature_initial (float or NDArray[np.float_]): Initial
+        temperature in Kelvin.
+        - pressure_initial (float or NDArray[np.float_]): Initial vapor
+        pressure in Pascals.
+        - temperature_final (float or NDArray[np.float_]): Final temperature
+        in Kelvin.
+
+        Returns:
+        - vapor_pressure_final (float or NDArray[np.float_]): Final vapor
+        pressure in Pascals.
+
+        References:
+        ----------
+        - https://en.wikipedia.org/wiki/Clausius%E2%80%93Clapeyron_relation
+        Ideal_gas_approximation_at_low_temperatures
+        """
+        return self.pressure_initial * np.exp(
+            (self.latent_heat / np.array(GAS_CONSTANT.m))
+            * (1 / self.temperature_initial - 1 / temperature))
+
+
+class WaterBuckStrategy(VaporPressureStrategy):
+    """Concrete implementation of the VaporPressureStrategy using the
+    Buck equation for water vapor pressure calculations."""
+
+    def pure_vapor_pressure(
+                self, temperature: Union[float, NDArray[np.float_]]
+            ) -> Union[float, NDArray[np.float_]]:
+        """
+        Calculate the pure (saturation) vapor pressure using the Buck
+        equation for water vapor.
+
+        Args:
+        ----
+        - temperature (float or NDArray[np.float_]): Temperature in Kelvin.
+
+        Returns:
+        -------
+        - vapor_pressure (float or NDArray[np.float_]): The vapor pressure in
+        Pascals.
+
+        References:
+        ----------
+        Buck, A. L., 1981: New Equations for Computing Vapor Pressure and
+        Enhancement Factor. J. Appl. Meteor. Climatol., 20, 1527-1532,
+        https://doi.org/10.1175/1520-0450(1981)020<1527:NEFCVP>2.0.CO;2.
+
+        https://en.wikipedia.org/wiki/Arden_Buck_equation
+        """
+        # Constants for the Buck equation
+        # Ensure input is numpy array for vectorized operations
+        temp = np.array(temperature) - 273.15  # Convert to Celsius
+        temp_below_freezing = temp < 0.0
+        temp_above_freezing = temp >= 0.0
+
+        # Buck equation separated for temperatures below and above freezing
+        vapor_pressure_below_freezing = 6.1115 * \
+            np.exp((23.036 - temp / 333.7) * temp / (279.82 + temp)) * \
+            100  # Convert from hPa to Pa
+        vapor_pressure_above_freezing = 6.1121 * \
+            np.exp((18.678 - temp / 234.5) * temp / (257.14 + temp)) * \
+            100  # Convert from hPa to Pa
+
+        # Select the appropriate equation based on temperature
+        vapor_pressure = (vapor_pressure_below_freezing * temp_below_freezing +
+                          vapor_pressure_above_freezing * temp_above_freezing)
+        return vapor_pressure
+
+
+def vapor_pressure_factory(
```

Comment:
The factory, will probably move to its own file either now, but forsure when we have an interface that hands building vapor_pressure strategies, that handles units and build methods and build validation checks.